cli/src/formatter.rs

```diff
@@ -522,6 +527,58 @@ impl<W: Write> Drop for ColorFormatter<W> {
     }
 }
 
+pub struct DebugFormatter<W: Write> {
+    formatter: ColorFormatter<W>,
+}
+
+impl<W: Write> DebugFormatter<W> {
+    pub fn new(output: W, rules: Arc<Rules>) -> Self {
+        Self {
+            formatter: ColorFormatter::new(output, rules),
+        }
+    }
+
+    pub fn for_config(output: W, config: &config::Config) -> Result<Self, config::ConfigError> {
+        let rules = rules_from_config(config)?;
+        Ok(Self::new(output, Arc::new(rules)))
+    }
+}
+
+impl<W: Write> Write for DebugFormatter<W> {
+    fn write(&mut self, data: &[u8]) -> Result<usize, Error> {
+        let labels = self.formatter.labels.clone();
+        write!(&mut self.formatter, "<<")?;
+        for (idx, label) in labels.iter().enumerate() {
+            if idx != 0 {
+                write!(&mut self.formatter, " ")?;
+            }
+            write!(&mut self.formatter, "{}", label)?;
+        }
```

Comment:
nit: it might be okay to simply use `{:?}` to dump the labels array (if we don't care much about the output.)

Reply:
That would be ok with me, I just figured it looks quite dense, I could probably also use something akin to `"{}", labels.iter().map(ToString::to_string).join(" ")`.

Reply:
Sounds good. `labels.join(" ")` should also work (and would be cheaper than cloning `Vec<String>`.